xarray/core/variable.py

```diff
@@ -944,7 +944,11 @@ def _shift_one_dim(self, dim, count):
             keep = slice(None)
 
         trimmed_data = self[(slice(None),) * axis + (keep,)].data
-        dtype, fill_value = dtypes.maybe_promote(self.dtype)
+
+        if fill_value is dtypes.NA and True:
+            dtype, fill_value = dtypes.maybe_promote(self.dtype)
+        else:
+            dtype = self.dtype
```

Comment:
What happens if filler is not compatible with self.dtype?
For example, feeding np.nan to an int array.
Probably it is a part of user responsibility and we do not need to take care of this, but I am just curious of it.

Reply:
In theory, NumPy should raise an error... But it may not.

Reply:
(this is the issue I'm looking at ref https://github.com/pydata/xarray/pull/2470#discussion_r223372119), good foresight @fujiisoup !